templates/commands/template_funcs.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"regexp"
+	"strings"
+)
+
+var (
+	// hyphenOrSnakeCaseKeep are all the characters to keep for generating snake_case or hyphen-case strings.
+	// The regex is used to match all characters not specified here so we can remove them.
+	hyphenOrSnakeCaseKeep = regexp.MustCompile(`[^a-zA-Z0-9-_ ]+`)
```

Comment:
Would these be cleaner as https://pkg.go.dev/strings#Replacer rather than regex?

Reply:
Never mind, this doesn't work for this use case.